modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/snapshot/GlobalSnapshotStateChannelManager.scala

```diff
@@ -0,0 +1,125 @@
+package org.tessellation.sdk.infrastructure.snapshot
+
+import cats.Eval
+import cats.data.{NonEmptyChain, NonEmptyList}
+import cats.effect.kernel.Async
+import cats.syntax.flatMap._
+import cats.syntax.functor._
+import cats.syntax.list._
+import cats.syntax.option._
+import cats.syntax.traverse._
+
+import scala.collection.immutable.SortedMap
+
+import org.tessellation.kryo.KryoSerializer
+import org.tessellation.schema.address.Address
+import org.tessellation.schema.{GlobalSnapshotInfo, SnapshotOrdinal}
+import org.tessellation.security.hash.Hash
+import org.tessellation.security.signature.Signed
+import org.tessellation.statechannel.{StateChannelOutput, StateChannelSnapshotBinary}
+import org.tessellation.syntax.sortedCollection._
+
+import _root_.cats.kernel.Order
+import eu.timepit.refined.auto._
+import eu.timepit.refined.types.numeric.PosLong
+import io.chrisdavenport.mapref.MapRef
+
+trait GlobalSnapshotStateChannelManager[F[_]] {
+  def accept(ordinal: SnapshotOrdinal, lastGlobalSnapshotInfo: GlobalSnapshotInfo, events: List[StateChannelOutput]): F[
+    (
+      SortedMap[Address, NonEmptyList[Signed[StateChannelSnapshotBinary]]],
+      Set[StateChannelOutput]
+    )
+  ]
+}
+
+object GlobalSnapshotStateChannelManager {
+  def make[F[_]: Async: KryoSerializer](ordinalDelay: Option[PosLong]): F[GlobalSnapshotStateChannelManager[F]] =
+    MapRef.ofConcurrentHashMap[F, (Address, Hash), SnapshotOrdinal]().map { firstSeenOrdinalR =>
+      new GlobalSnapshotStateChannelManager[F] {
+
+        override def accept(ordinal: SnapshotOrdinal, lastGlobalSnapshotInfo: GlobalSnapshotInfo, events: List[StateChannelOutput]): F[
```

Comment:
there is nothing to override as you extend nothing but implement a new implementation